openapi.yaml

```diff
@@ -1016,18 +1016,30 @@ components:
           properties:
             code:
               enum: [NOSU]
-    JobPostAttributes:
+    JobSuffixRequired:
       type: object
-      additionalProperties: false
       properties:
         suffix:
           type: string
           nullable: false
           maxLength: 2
           description: >-
             Used to uniquely identify a job record for persons who have multiple
-            jobs that share the same position number. For updating existing
-            jobs, this field is required.
+            jobs that share the same position number. This field is required.
+    JobSuffixOptional:
```

Comment:
Why do we have two suffixes attribute defined and the only difference is `nullable: true/false`? I think what you should do are the following:

1. Make suffix `nullable: true` in post attributes.
2. Separate patch attributes from post attributes (you can still reuse common attributes by references).
3. Remove suffix from patch attributes.
4. Also make sure you modify the API code body including unit tests to reflect the change. For example, we will no longer need to check whether suffixes are matched between path and body for patch.